### PR TITLE
Added metrics support

### PR DESCRIPTION
**Pull request by empijei**

### CODE REVIEW

File: proxy/internal/specialized/metrics.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+	// Miss is the overall amount of misses, which means the requested key was not in MFA nor in LRU.
+	Miss uint
+
+	// RecentlyEvictedMiss is the amount of misses that had a key which was recently evicted.
+	// If the cache is not running with evicMetrics it will be set to 0.
+	RecentlyEvictedMiss uint
+}
+
+// Hits returns the total amount of hits.
```
Comment (mikispag) [inline proxy/internal/specialized/metrics.go]: ``Hit``

File: proxy/cache.go [hunk 2]
```diff
@@ -20,7 +20,7 @@ type cacheValue struct {
 }
 
 func newCache(size int) (*cache, error) {
-	c, err := specialized.NewCache(size)
+	c, err := specialized.NewCache(size, true)
```
Comment (mikispag) [inline proxy/cache.go]: Can we make this a flag with a default value of false?

Comment (empijei) [reply proxy/cache.go]: Done.

File: proxy/internal/specialized/metrics.go (hunk 1 above)
Comment (empijei) [reply proxy/internal/specialized/metrics.go]: Done.

**Comment by empijei** [comment]
Addressed nits and squashed commits

File: main.go [hunk 3]
```diff
@@ -20,6 +20,7 @@ var (
 	upstreamServers = flag.String("s", "one.one.one.one:853@1.1.1.1,dns.google:853@8.8.8.8", "comma-separated list of upstream servers")
 	logPath         = flag.String("l", "", "log file path")
 	isLogVerbose    = flag.Bool("v", false, "verbose mode")
+	evictMetrics    = flag.Bool("em", false, "collect metrics on evictions")
```
Comment (mikispag) [inline main.go]: Please update README, section Usage.

Comment (empijei) [reply main.go]: Done, even if copy-pasting the output of the "-help" flag in the readme seems quite redundant.

**Comment by mikispag** [comment]
Thank you so much!

**Pull request merged**